cdc/log.cc

```diff
@@ -722,8 +728,20 @@ cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout,
 
             transformer trans(_ctxt, s);
 
+            std::vector<mutation> muts;
+            if (should_split(m, *s)) {
+            } else {
+                muts.push_back(std::move(m));
+            }
+
             if (!s->cdc_options().preimage()) {
-                mutations.emplace_back(trans.transform(m));
+                if (should_split(m, *s)) {
+                    for(auto&& mm : split(m, s)) {
+                        mutations.emplace_back(trans.transform(mm));
+                    }
+                } else {
+                    mutations.emplace_back(trans.transform(m));
+                }
```

Comment:
Again, code repetition. Why not just make a ready future `<result_set null>`, and set it to result of call iff preimage is enabled?

Reply:
Good idea, will do.